feat: download correct binary when in FedRAMP environment

adamconnelly opened the pull request:
## Description of the change

I've updated the startup script to check whether it's connecting to the MQTT broker for our FedRAMP environment. If so it downloads the FedRAMP-specific version of the launcher.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue);
- [x] New feature (non-breaking change that adds functionality);
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected);
- [ ] Documentation (a documentation or example fix not affecting the infrastructure managed by this module);

## Checklists

### Development

- [x] All necessary variables have been defined, with defaults if applicable;
- [x] The code is formatted properly;

### Code review

- [x] The module version is bumped accordingly;
- [x] Spacelift tests are passing;
- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached;
- [x] This pull request is no longer marked as "draft";
- [x] Reviewers have been assigned;
- [x] Changes have been reviewed by at least one other engineer;

jmfontaine reviewed (dismissed):
👍

jmfontaine commented on README.md:
```diff
@@ -56,6 +56,49 @@ module "spacelift_workerpool" {
+}
+
+module "spacelift_workerpool" {
+  source = "github.com/spacelift-io/terraform-aws-spacelift-workerpool-on-ec2?ref=v4.4.3"
```
This should be updated to use the upcoming `5.3.0` tag. The main example should be updated too.

jmfontaine commented on README.md:
```diff
@@ -56,6 +56,49 @@ module "spacelift_workerpool" {
 
+### FedRAMP Example
+
+If your account is in the Spacelift FedRAMP environment, set the `is_fedramp` variable to `true`:
```
@adamconnelly On second thought, do we need the whole example or just this one line. We might want to have a full working example in the `examples` folder instead of cluttering the README file.

Apollorion commented on user_data/saas.tftpl:
```diff
@@ -12,7 +12,7 @@ spacelift () {(
   fi
 
-  baseURL="https://downloads.${domain_name}/spacelift-launcher"
+  baseURL="https://downloads.${domain_name}/spacelift-launcher${is_fedramp ? "-fedramp" : ""}"
```
or even, if we have different dns for different customers use like contains("spacelift.gov") or something like that?

adamconnelly replied:
Yes - great idea. I was just a little nervous about basing it on domain name, but in reality it's unlikely to change.

One thing, we can't use the domain name variable because both binaries are downloaded from the same domain (downloads.spacelift.io), but I think what we could do is decode the token and try to find out what MQTT domain is being connected to. I'll try to do that.

adamconnelly commented:
@jmfontaine @Apollorion @frncmx I've updated this to be a bit simpler based on Joey's suggestion. Instead of a new var, I'm just looking at the MQTT broker hostname to decide which binary to download. This means it'll just work seamlessly for customers with no extra configuration for FedRAMP.

Apologies that I squashed all the changes instead of pushing separate commits - I just did it by accident without thinking.

Apollorion reviewed (approved):
<img src="https://media0.giphy.com/media/v1.Y2lkPWJkM2VhNTdlMGI3ZmE1dGx2ZmFrbXhheTdkZ2Vrc3JnYzBsejhjNmt3eTJodnNrNCZlcD12MV9naWZzX3NlYXJjaCZjdD1n/KXY5lB8yOarLy/giphy.gif"/>
